test/verify/netlib.py

```diff
@@ -29,9 +30,18 @@ def setUp(self):
         # connections that might still be here from the time of
         # creating the image and we prevent NM from automatically
         # creating new connections.
+        # if the command fails, try again
+        failures_allowed = 3
+        while True:
+            try:
+                print m.execute("nmcli con show")
+                m.execute("""nmcli -f UUID,DEVICE connection show | awk '$2 == "--" { print $1 }' | xargs -r nmcli con del""")
+                break
+            except subprocess.CalledProcessError:
+                failures_allowed -= 1
+                if failures_allowed == 0:
+                    raise
```

Comment:
For the future:  This can be written a bit more concisely with
```
for retry in range(3, -1, -1):
    try:
        subprocess.check_call(['true'])
        break
    except subprocess.CalledProcessError as e:
        if retry == 0:
            raise
```
But that's utter nitpicking. LGTM now, thanks! Let's wait until some tests go green, then I'll merge.